ledger/simulation/tracer.go

```diff
@@ -200,6 +294,17 @@ func (tracer *evalTracer) AfterOpcode(cx *logic.EvalContext, evalError error) {
 	tracer.handleError(evalError)
 }
 
+func (tracer *evalTracer) BeforeProgram(cx *logic.EvalContext) {
+	// Before Program, activated for logic sig, happens before txn group execution
+	// we should create trace object for this txn result
+	if cx.RunMode() != logic.ModeApp {
+		if tracer.result.ExecTraceConfig > NoExecTrace {
+			indexIntoTxnGroup := cx.GroupIndex()
+			tracer.result.TxnGroups[0].Txns[indexIntoTxnGroup].Trace = &TransactionTrace{}
```

Comment:
Are you re-using the first Txn's trace to hold a logic sig trace?

Reply:
Not exactly, I am creating a trace for transaction result to hold logis sig trace, rather than re-using it.

This is for our implementation verify logic sig before executing txn-group, then txn-result's trace is not there yet. Thus, we have to create an empty transaction-trace to hold the logic sig trace.

For more info, I left what I thought in a discussion with Jason here: https://github.com/algorand/go-algorand/pull/5360#discussion_r1195207209